bring back old Hcal digi/reco chain for testing/debug

tomeichlersmith opened the pull request:
This is a transfer of https://github.com/LDMX-Software/Hcal/pull/75 into this repository.

I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
This resolves #1351 

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments
- [x] I ran my developments and the following shows that they are successful. See https://github.com/LDMX-Software/Hcal/pull/75

EinarElen committed:
bring back old Hcal digi/reco chain for testing/debug

This is a transfer of https://github.com/LDMX-Software/Hcal/pull/75
into this repository.

github-actions[bot] committed:
Apply clang-format

tvami commented on Hcal/src/Hcal/HcalSimpleDigiAndRecProducer.cxx:
```diff
@@ -0,0 +1,149 @@
+    recHit.setSection(hitID.section());
+    recHit.setStrip(hitID.strip());
+    recHit.setLayer(hitID.layer());
+    recHit.setEnergy(edep);
```
Can we add the `setOrientation` function like I did in https://github.com/LDMX-Software/ldmx-sw/pull/1577/files please?

tvami commented on Hcal/include/Hcal/HcalSimpleDigiAndRecProducer.h:
```diff
@@ -0,0 +1,40 @@
+      : framework::Producer{name, process} {}
+  ~HcalSimpleDigiAndRecProducer() override = default;
+  void configure(framework::config::Parameters& ps) override;
+  void SetupRandomNumberGeneration();
```
shouldnt this be lower case, since it's a function, not a class?

tomeichlersmith replied:
yep! and its content can be moved to `onNewRun` after a Framework patch awhile ago

tvami commented on Hcal/src/Hcal/HcalSimpleDigiAndRecProducer.cxx:
```diff
@@ -0,0 +1,149 @@
+
+void HcalSimpleDigiAndRecProducer::configure(
+    framework::config::Parameters& ps) {
+  input_coll_name = ps.getParameter<std::string>("input_coll_name");
+  input_pass_name = ps.getParameter<std::string>("input_pass_name");
+  output_coll_name = ps.getParameter<std::string>("output_coll_name");
+  mev_per_mip = ps.getParameter<double>("mev_per_mip");
+  pe_per_mip = ps.getParameter<double>("pe_per_mip");
+  attenuation_length = ps.getParameter<double>("attenuation_length");
+  readout_threshold = ps.getParameter<int>("readout_threshold");
+  mean_noise = ps.getParameter<double>("mean_noise");
+  position_resolution = ps.getParameter<double>("position_resolution");
```
also if we follow our (not-yet-written-down) coding rules, these should end with a trailing underscore, showing that these are global variables

tomeichlersmith replied:
will do :+1:

tomeichlersmith committed:
resolve Tamas review comments and switch to STL

- use STL distributions and RNG instead of TRandom3
- add underscores after member variables to distinguish from local
  variables
- move seeding of RNGs into onNewRun instead of at the begninning of the
  first event

tvami commented:
> switched from TRandom3 to the STL distributions/generator so that we have a concrete example of what I outlined in https://github.com/LDMX-Software/ldmx-sw/issues/1388#issuecomment-2286459236

This is great! I think we can resolve that next after we cut a gold with some of the recent changes
